### PR TITLE
feature/delete-category

### PR DESCRIPTION
**Pull request by Jifflis**
# 📲 What #
- Create delete category api
 
# 🛠 How #
- Create delete category mutation

# ✅ Acceptance criteria #
- Create Test
 
# ⏰ TODO #
None

# 🔗 References #
https://trello.com/c/MIVQ9xqN/32-delete-category

### CODE REVIEW

File: app/interactors/delete_category.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+# app/interactors/delete-category.rb
+class DeleteCategory
+  include Interactor
+
+  def call
+    @current_user = context[:current_user]
```
Comment (jaimejazarenoiii) [inline app/interactors/delete_category.rb]: `@`

File: test/graphql/mutations/delete_category_test.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+             query: sign_in_with_email_mutation,
+             variables: sign_in_with_email_mutation_variables({ email: @user.email,
+                                                                password: password })
+           })
+      @category = @user.categories.first
+      @json_response = parse_graphql_response(response.body)
+      @token = @json_response.dig('signInWithEmail', 'token')
+    end
+
+    test 'valid delete category' do
```
Comment (jaimejazarenoiii) [inline test/graphql/mutations/delete_category_test.rb:25]: can we test the failures also? Like record not found

File: test/support/graphql/mutations_helper.rb [hunk 3]
```diff
@@ -146,6 +146,30 @@ def delete_wallet_mutation(_input = {})
         }
       )
     end
+
+    def delete_category_mutation(_input = {})
+      %(
+        mutation DeleteCategory(
+          $id: ID!,
```
Comment (jaimejazarenoiii) [inline test/support/graphql/mutations_helper.rb]: no need for commas?

File: test/support/graphql/mutations_helper.rb [hunk 4]
```diff
@@ -146,6 +146,30 @@ def delete_wallet_mutation(_input = {})
...
+        ) {
+          deleteCategory(input: {
+            id: $id,
```
Comment (jaimejazarenoiii) [inline test/support/graphql/mutations_helper.rb]: same here

**Review by jaimejazarenoiii — CHANGES_REQUESTED** [review]

**Review by rexendy — APPROVED** [review]

**Pull request merged**